File: LeetCode_2154/Main.py
from typing import List

class Solution:
    def findFinalValue(self, nums: List[int], original: int) -> int:
        while original in nums:
            original *= 2
        return original

print(Solution().findFinalValue(nums = [5,3,6,1,12], original = 3))
print(Solution().findFinalValue(nums = [2,7,9], original = 4))

File: LeetCode_2154/Main.cpp
#include<bits/stdc++.h>
using namespace std;

class Solution {
public:
    int findFinalValue(vector<int>& nums, int original) {
        vector<int> cnt(1001, 0);
        for (int x : nums) {
            cnt[x] += 1;
        }

        while (original < 1001 and cnt[original]) {
            original *= 2;
        }
        return original;
    }
};